File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/rust/handlers.rs
use crate::service::{create_dog, create_cat, transfer_animal, list_animals};
use crate::models::{Dog, Cat, Shelter};

pub fn handle_create_dog(name: String) -> Dog {
    create_dog(name)
}

pub fn handle_create_cat(name: String) -> Cat {
    create_cat(name)
}

pub fn handle_transfer(name: String) -> usize {
    let mut shelter = Shelter::new();
    let dog = Dog::new(name);
    transfer_animal(dog.name, &mut shelter);
    shelter.count()
}

pub fn handle_list() -> Vec<String> {
    let animals = list_animals();
    animals
}

pub fn validate(name: &str) -> bool {
    !name.is_empty()
}

File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/go/handlers.go
package handlers

import (
	"myapp/models"
	"myapp/service"
)

func HandleCreateDog(name string) *models.Dog {
	return service.CreateDog(name)
}

func HandleCreateCat(name string) *models.Cat {
	return service.CreateCat(name)
}

func HandleTransfer(name string) int {
	shelter := models.NewShelter()
	dog := models.NewDog(name)
	service.TransferAnimal(dog.Name, shelter)
	return shelter.Count()
}

func HandleList() []string {
	animals := service.ListAnimals()
	return animals
}

func Validate(name string) bool {
	return len(name) > 0
}
